src/library.js

```diff
@@ -1238,13 +1238,33 @@ mergeInto(LibraryManager.library, {
     return _strptime(buf, format, tm); // no locale support yet
   },
 
+  // Base Emscripten EH error class
+#if TRANSPILE_TO_ES5
+  // FIXME: manually transpile due to https://github.com/google/closure-compiler/issues/1138
+  $EmscriptenEH__docs: '/** @constructor */',
+  $EmscriptenEH: 'function EmscriptenEH() {}\n' +
+                 'EmscriptenEH.prototype = Error.prototype;',
+#else
+  $EmscriptenEH: 'class EmscriptenEH extends Error {}',
+#endif
+
   // ==========================================================================
   // setjmp.h
   // ==========================================================================
 
 #if SUPPORT_LONGJMP == 'emscripten'
+  $EmscriptenSjLj__deps: ['$EmscriptenEH'],
+#if TRANSPILE_TO_ES5
+  // FIXME: manually transpile due to https://github.com/google/closure-compiler/issues/1138
```

Comment:
Sadly our transpilation strategy doesn't work when closure-compiler requires helper functions.  This is because we want to be able to run it in `WHITESPACE_ONLY` mode, but with transpilation enabled, which works, but it doesn't inject its helper functions.

I think in the long term we would need to switch to a new transpiler to make more advanced stuff work.  Even things like `for .. of` don't currently work.

Reply:
For simplicity maybe just skip this `if/else` completely and revert the `TRANSPILE_TO_ES5` changes?

Reply:
Commit 9b5044844c6e1e9798944625c847d13eb230818e moves this behind a setting and commit 2cc75f3a178d11952f0f29a011b53f146aef0c67 disabled that setting when ES6 classes are not supported.